### PR TITLE
[JENKINS-68126] Remove event streaming causing performance problems at scale

### PR DESCRIPTION
**Pull request by Vlatombe**
Amends #1167 

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java [hunk 1]
```diff
@@ -166,12 +165,12 @@ else if (httpCode == 409 && e.getMessage().contains("Operation cannot be fulfill
...
-            watcher = new AllContainersRunningPodWatcher(client, pod);
-            try (Watch w1 = client.pods().inNamespace(namespace).withName(podName).watch(watcher);
-                 Watch w2 = eventWatch(client, podName, namespace, runListener)) {
-                assert watcher != null; // assigned 3 lines above
-                watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+
+            Pod checkPod = client.pods().inNamespace(namespace).withName(podName).waitUntilReady(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+            if (!Readiness.isPodReady(checkPod)) {
+                throw new KubernetesClientTimeoutException(pod, template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
             }
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java:172]: Did we establish whether `waitUntilReady` throws an exception if it times out, in which case
```suggestion
            client.pods().inNamespace(namespace).withName(podName).waitUntilReady(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
```
?

**Review by jglick — APPROVED** [review]
Seems to involve a loss of some diagnosability, but that seems an acceptance price to pay for the scalability improvement (and code simplification).

**Comment by jglick** [comment]
Note that automatic release after matching labels is currently not reliable, possibly due to a race condition in GitHub (being investigated); if you wish, you can force a release with **Run workflow** at https://github.com/jenkinsci/kubernetes-plugin/actions/workflows/cd.yaml.

**Comment by MattLud** [comment]
> Seems to involve a loss of some diagnosability, but that seems an acceptance price to pay for the scalability improvement (and code simplification).

This removal puts us in a bind for providing user feedback on builds. 

Previously, this would inform them that their pod wasn't scheduled due to lack of resources or other capacity issues. 

Now we're back to a "spinning beachball" setup - a user needs to ping platform owners to investigate why a pod isn't running.

**Comment by jglick** [comment]
It would be great to readd support for streaming agent pod event messages (and also pod logs) to the build log, if this can be done without burdening either the controller or the API server.

**Pull request merged**